Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -3999,6 +4005,19 @@ void CGame::Packet_DiscordJoin(CDiscordJoinPacket& Packet)
     }
 }
 
+void CGame::Packet_PlayerResourceStart(CPlayerResourceStartPacket& Packet)
+{
+    CPlayer* pPlayer = Packet.GetSourcePlayer();
+    if (pPlayer)
+    {
+        CLuaArguments Arguments;
+        CResource*    pResource = Packet.GetResource();
+
+        Arguments.PushResource(Packet.GetResource());
+        pPlayer->CallEvent("onPlayerResourceStart", Arguments, NULL);
```

Comment:
Also I think you should check if `GetResource()` a non-null value. It might happen (due to packet loss) the packet is recieved late, and by that point the resource has been deleted server side. Rare, but it could happen.

Reply:
And if `GetResource()` is null? Just don't call the event?

Reply:
Well, calling the event wouldn't make a lot of sense, since it would become nil in the end.